### DISCUSSION TITLE
Best way to initialize Sharepreferences using code generation

### DISCUSSION THREAD

**Discussion by drinikol** [rrousselGit/riverpod#3726] [Q&A]
Previously I was using this, but now I am trying to convert it to the code generation but apparently it always fails or throws an the UnimplementedError. Hopefully somebody can help me how to make if work. 

Previous.
``` 
final sharedPreferencesProvider = Provider<SharedPreferences>((ref) {
  throw UnimplementedError();
});

```
New
``` 
@riverpod
SharedPreferences sharedPrefs(SharedPrefsRef ref) {
  throw UnimplementedError();
} 

Future<void> main() async {
  final SharedPreferences prefs = await SharedPreferences.getInstance();
  return runApp(ProviderScope(
    overrides: [
      sharedPreferencesProvider.overrideWithValue(prefs),
    ],
    child: MyApp(),
  ));
} 
```

**Comment by rrousselGit** [comment]
Your code should work. The issue is most likely elsewhere and unrelated to codegen vs non-codegen

**Reply by drinikol** [reply]
@rrousselGit thank you for your response! tried it in a fresh project and it worked. I need to refactor the files properly.
